core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/OpenSearchItem.kt

```diff
@@ -32,12 +32,12 @@ data class OpenSearchItem(
   private val openInNewTab: Boolean = false
 ) : SideEffect<Unit> {
   override fun invokeWith(activity: AppCompatActivity) {
-    activity.setNavigationResult(
+    val readerFragmentResId = (activity as CoreMainActivity).readerFragmentResId
+    activity.navigate(readerFragmentResId)
+    activity.setNavigationResultOnCurrent(
```

Comment:
@MohitMaliFtechiz It's the same process that we were using previously, instead of previous we are now setting `currentBackStackEntry`. I think this problem will happen again. Am i missing something?

Reply:
hi @gouri-panda ,  Please explain  more on this while the test requirement was to start new Reader fragment  after searching the text from search fragment . Since its new reader fragment so we set all object of Bundle on it.

Reply:
I was saying that in the previous we set `previousNavBackStackEntry` when we first navigate and then set the value. Now we are setting `currentNavBackStackEntry` when we first set the value and then we navigate, which is basically the same. So I think the change is unnecessary.

Reply:
Hi @gouri-panda ,

It should be same but i tried previous one and that was not working , search is not working with that code, so i have change the order.